src/wallet/wallet.h

```diff
@@ -269,7 +288,24 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
     bool fBackupMints;
     std::unique_ptr<CzPIVTracker> zpivTracker;
 
-    std::set<int64_t> setKeyPool;
+   void LoadKeyPool(int nIndex, const CKeyPool &keypool)
+    {
+        if (keypool.fInternal) {
+            setInternalKeyPool.insert(nIndex);
+        } else {
+            setExternalKeyPool.insert(nIndex);
+        }
+
+        // If no metadata exists yet, create a default with the pool key's
+        // creation time. Note that this may be overwritten by actually
+        // stored metadata for that key later, which is fine.
+        CKeyID keyid = keypool.vchPubKey.GetID();
+        if (mapKeyMetadata.count(keyid) == 0)
+            mapKeyMetadata[keyid] = CKeyMetadata(keypool.nTime);
+    }
+
+    std::set<int64_t> setInternalKeyPool;
+    std::set<int64_t> setExternalKeyPool;
```

Comment:
I think that this is not decoupled properly. 

Would be better suited inside the hd chain object or an intermediary hd family object, for example:

Wallet --> HD chain ("family") --> Set of simple chains (the current HDChain object).